microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt

```diff
@@ -0,0 +1,176 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Bookmark
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.AnimationCurve
+import com.arcgismaps.toolkit.geocompose.MapView
+import com.arcgismaps.toolkit.geocompose.MapViewpointOperation
+
+/**
+ * Displays a composable [com.arcgismaps.toolkit.geocompose.MapView] and permits setting the viewpoint
+ * using options in a dropdown menu. The dropdown menu options represent different methods of setting
+ * the viewpoint, and all of the center on the same location. A circular progress indicator is displayed
+ * over the map while an operation is in progress.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen() {
+    val arcGISMap by remember { mutableStateOf(ArcGISMap(BasemapStyle.ArcGISStreets)) }
+    var mapViewpointOperation: MapViewpointOperation? by remember {
+        mutableStateOf(null)
+    }
+    var showProgressIndicator by remember { mutableStateOf(false) }
+    LaunchedEffect(key1 = mapViewpointOperation) {
+        showProgressIndicator = true
+        mapViewpointOperation?.await()
+        showProgressIndicator = false
+    }
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                colors = TopAppBarDefaults.topAppBarColors(
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    titleContentColor = MaterialTheme.colorScheme.primary,
+                ),
+                title = { Text("MapView Set Viewpoint App") },
+                actions = {
+                    var actionsExpanded by remember { mutableStateOf(false) }
+                    IconButton(onClick = { actionsExpanded = !actionsExpanded }) {
+                        Icon(Icons.Default.MoreVert, "More")
+                    }
+
+                    SetViewpointDropdownMenu(
+                        onSetViewpointOperation = {
+                            mapViewpointOperation = it
+                        },
+                        expanded = actionsExpanded
+                    ) {
+                        actionsExpanded = false
+                    }
```

Comment:
I would be more in favor of using the named argument here, as it shows when is the actionsExpanded being set to false:

```suggestion
                    SetViewpointDropdownMenu(
                        onSetViewpointOperation = { mapViewpointOperation = it },
                        expanded = actionsExpanded,
                        onDismissRequest = {
                            actionsExpanded = false
                        }
                    )
```